bundle/edu.gemini.horizons.api/src/main/java/edu/gemini/horizons/server/backend/CgiQueryExecutor.java

```diff
@@ -60,80 +53,68 @@ private static synchronized String _formatDate(Date date) {
      * @return the reply from the Horizons' service for the given query
      */
-    public HorizonsReply execute(HorizonsQuery query) throws HorizonsException {
-        final List<NameValuePair> queryParams = _initQueryParams();
+    public HorizonsReply execute(final HorizonsQuery query) throws HorizonsException {
+        final Map<String, String> queryParams = _initQueryParams();
         _initSite(query, queryParams);
         _initDateParams(query, queryParams);
 
-        String objectId = query.getObjectId().trim();
-        //For asteroids and minor planets, add ";" at the end of the query. This
-        //makes the JPL/Horizons to narrow the results.
-        if (query.getObjectType() == HorizonsQuery.ObjectType.MINOR_BODY) {
-            objectId += ";"; //
-        }
-
         //if the object id contains spaces, surround the id in ' '
-        if (objectId.contains(" ")) {
-            StringBuffer sb = new StringBuffer();
-            sb.append("'");
-            sb.append(objectId);
-            sb.append("'");
-            objectId = sb.toString();
-        }
+        final String objectId = query.getObjectId().trim();
+        queryParams.put(CgiHorizonsConstants.COMMAND, objectId.contains(" ") ? "'" + objectId + "'" : objectId);
 
-        queryParams.add(new NameValuePair(CgiHorizonsConstants.COMMAND, objectId));
         if (query.getStepSize() > 0) {
-            queryParams.add(new NameValuePair(CgiHorizonsConstants.STEP_SIZE, query.getStepSize() + query.getStepUnits().suffix));
+            queryParams.put(CgiHorizonsConstants.STEP_SIZE, query.getStepSize() + query.getStepUnits().suffix);
         }
 
-        final GetMethod method = new GetMethod(CgiHorizonsConstants.HORIZONS_URL);
-        method.setQueryString(_buildParameterList(queryParams));
-        HorizonsReply reply;
-        HttpClient client = new HttpClient();
         try {
-            client.executeMethod(method);
-            reply = CgiReplyBuilder.buildResponse(method.getResponseBodyAsStream(), method.getRequestCharSet());
-            method.releaseConnection();
-        } catch (IOException ex) {
+            // Build the parameter string and URL.
+            final String charSet = "UTF-8";
+            final StringBuilder queryParamString = new StringBuilder();
+            boolean first = true;
+            for (final Map.Entry<String, String> e: queryParams.entrySet()) {
+                if (first) first = false;
+                else queryParamString.append("&");
+                queryParamString.append(e.getKey()).append("=").append(URLEncoder.encode(e.getValue(), charSet));
+            }
+            final URL url = new URL(CgiHorizonsConstants.HORIZONS_URL + "?" + queryParamString);
+            LOG.info("Horizons request " + url);
+
+            final HttpsURLConnection conn = (HttpsURLConnection) url.openConnection();
+            conn.setHostnameVerifier((v1,v2) -> true);
```

Comment:
There's no reason why I couldn't rewrite `CgiQueryExecutor` in scala, but at that point, it feels like it starts to pseudo-move to the solution for REL-3111.